mlir/lib/Target/IRDLToCpp/IRDLToCpp.cpp

```diff
@@ -0,0 +1,530 @@
+          [&dict, &perOpDefTemplate =
+                      perOpDefTemplate](irdl::OperationOp op) -> std::string {
+            auto opStrings = getStrings(op);
+            fillDict(dict, opStrings);
+
+            const auto operandCount = opStrings.opOperandNames.size();
+            const auto operandNames =
+                operandCount ? joinNameList(opStrings.opOperandNames)
+                             : "{\"\"}";
+
+            const auto resultNames = joinNameList(opStrings.opResultNames);
+
+            const auto buildDefinition = llvm::formatv(
+                R"(
+void {0}::build(::mlir::OpBuilder &odsBuilder, ::mlir::OperationState &odsState, {1} {2} ::llvm::ArrayRef<::mlir::NamedAttribute> attributes) {{
+{3}
+{4}
+}
+    )",
+                opStrings.opCppName,
+                llvm::join(llvm::map_range(opStrings.opResultNames,
+                                           [](StringRef attr) -> std::string {
+                                             return llvm::formatv(
+                                                 "::mlir::Type {0}, ", attr);
+                                           }),
+                           ""),
+                llvm::join(llvm::map_range(opStrings.opOperandNames,
+                                           [](StringRef attr) -> std::string {
+                                             return llvm::formatv(
+                                                 "::mlir::Value {0}, ", attr);
+                                           }),
+                           ""),
+                llvm::join(llvm::map_range(opStrings.opOperandNames,
+                                           [](StringRef attr) -> std::string {
+                                             return llvm::formatv(
+                                                 "  odsState.addOperands({0});",
+                                                 attr);
+                                           }),
+                           "\n"),
+                llvm::join(llvm::map_range(opStrings.opResultNames,
+                                           [](StringRef attr) -> std::string {
+                                             return llvm::formatv(
+                                                 "  odsState.addTypes({0});",
+                                                 attr);
+                                           }),
+                           "\n"));
+            dict["OP_BUILD_DEFS"] = buildDefinition;
+
+            std::string str;
+            llvm::raw_string_ostream stream{str};
+            perOpDefTemplate.render(stream, dict);
+            return str;
+          }),
+      "\n");
+
+  dict["OP_LIST"] = commaSeparatedOpList;
+  dict["OP_CLASSES"] = perOpDefinitions;
+  opDefTemplate.render(output, dict);
+  dialectDefTemplate.render(output, dict);
+
+  output << "#endif // " << definitionMacroFlag << "\n";
+  return success();
+}
+
+static LogicalResult verify(irdl::DialectOp dialect) {
```

Comment:
Just so people don't get the wrong idea (`verify` functions are very connotative in MLIR).
```suggestion
static LogicalResult verifySupported(irdl::DialectOp dialect) {
```